### DISCUSSION TITLE
Will PrimeNG 18 go along with an updated Figma UI kit?

### DISCUSSION THREAD

**Discussion by driekwartsbroek** [primefaces/community#2898] [Figma UI Kit] [answered]
I'm asking since we're a large enterprise that recently decided to go for PrimeNG. I'm an in-house UX designer, and unsure about the next steps. Should we await a new Figma file with a new token structure that will go along with the new design tokens in v18, or should I already start customising on the current UI kit and hope the discrepancy in token semantics between the current UI kit version and the new one isn't too big? 

I read this line in the UI Kit currently: "The tokens are named for code compatibility. Please note that token names will undergo significant changes soon due to the new theme engine." So should I await the new tokens?

**Comment by cagataycivici** [comment, marked as answer]
Yes, as a matter of fact the release of both the PrimeNG v18 and the UI kit v3 are aligned. We'll also offer a Figma plugin to generate PrimeNG themes from Figma after the release. The reference in the second paragraph of your message has been done. We plan to do a release for UI Kit next week along with PrimeNG v18 rc.1.

Edit: Typo fix | UI Kit v4 ➡️ UI Kit v3

**Reply by driekwartsbroek** [reply]
Thats fantastic news, thanks for the quick update!

**Reply by GuillaumeWenabi** [reply]
Hello, what a great news ! Is the UI kit v4 already released ? If not, do you know when it'll be available ?

**Reply by carol-braileanu** [reply]
any news on this update?

**Comment by cagataycivici** [comment]
PrimeNG v18 and v19 are fully compatible with UI Kit v3

**Reply by megaroeny** [reply]
I was wondering about that. What is the main difference with v18 vs v19? I don't see a v19 on the PrimeNG website.

**Comment by chodgesfigma** [comment]
Hey PrimeNG, with your latest UI kit in Figma. Is Figma's code connect something that you are looking to support with the new UI kit file? [Code Connect](https://github.com/figma/code-connect)

**Reply by onursenture** [reply]
Hi, as of now, we do not have any plans regarding Code Connect. We are monitoring developments on Figma side.

**Comment by diego-vannucci** [comment]
Hi, is there any news about the Figma plugin to generate PrimeNG themes?

**Reply by cagataycivici** [reply]
We plan to release it next week, currently at QA testing phase before launch.
